projects/ngx-maplibre-gl/src/lib/control/scale-control.directive.ts

```diff
@@ -1,38 +1,65 @@
 import {
-  AfterContentInit,
   Directive,
-  Host,
-  Input,
   OnChanges,
   SimpleChanges,
+  afterNextRender,
+  inject,
+  input,
 } from '@angular/core';
-import { ScaleControl, ScaleControlOptions } from 'maplibre-gl';
+import { ScaleControl,  type ScaleControlOptions } from 'maplibre-gl';
 import { MapService } from '../map/map.service';
 import { ControlComponent } from './control.component';
 
 /**
  * `mglScale` - a scale control directive
- * 
+ *
  * @category Directives
- * 
+ *
  * @see [Scale](https://maplibre.org/ngx-maplibre-gl/demo/ngx-scale-control)
  * @see [ScaleControl](https://maplibre.org/maplibre-gl-js/docs/API/classes/ScaleControl)
  */
 @Directive({
   selector: '[mglScale]',
   standalone: true,
 })
-export class ScaleControlDirective implements AfterContentInit, OnChanges {
+export class ScaleControlDirective implements OnChanges {
+  /* Init injection */
+  private readonly mapService = inject(MapService);
+  private readonly controlComponent = inject<ControlComponent<ScaleControl>>(
+    ControlComponent,
+    { host: true }
+  );
+
   /* Init inputs */
-  @Input() maxWidth?: number;
+  readonly maxWidth = input<number>();
 
   /* Dynamic inputs */
-  @Input() unit?: 'imperial' | 'metric' | 'nautical';
+  readonly unit = input<'imperial' | 'metric' | 'nautical'>();
+
+  constructor() {
+    afterNextRender(() => {
+      this.mapService.mapCreated$.subscribe(() => {
+        if (this.controlComponent.control) {
+          throw new Error('Another control is already set for this control');
+        }
+        const options: ScaleControlOptions = {};
+        const maxWidth = this.maxWidth();
+        if (maxWidth !== undefined) {
+          options.maxWidth = maxWidth;
+        }
+        const unit = this.unit();
 
-  constructor(
-    private mapService: MapService,
-    @Host() private controlComponent: ControlComponent<ScaleControl>
-  ) {}
+        if (unit !== undefined) {
+          options.unit = unit;
+        }
+        this.controlComponent.control = new ScaleControl(options);
+        this.mapService.addControl(
+          this.controlComponent.control,
+          this.controlComponent.position()
+        );
+      });
+    });
+  }
```

Comment:
What about ngonchanges? It will remain as is?

Reply:
Some onChanges can be replaced by signal effects but I think it would be better to do that in a PR because it would require even more modifications